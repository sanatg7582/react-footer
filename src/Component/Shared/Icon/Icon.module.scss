.iconImage {
  padding: 5px;
}
