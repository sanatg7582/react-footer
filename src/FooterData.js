const footerData = [
  {
    headingId: 1,
    heading: "Customer Service",
    Items: [
      { ItemId: 1, title: "Help Center" },
      { ItemId: 2, title: "Contact Us" },
      { ItemId: 3, title: "Return an Item" },
      { ItemId: 4, title: "Recall Information" },
      { ItemId: 5, title: "Safety Data Sheets" }
    ]
  },
  {
    headingId: 2,
    heading: "Corporate Info",
    Items: [
      { ItemId: 1, title: "Stapies Business Advantage Overview" },
      { ItemId: 2, title: "Industries Served" },
      { ItemId: 4, title: "Blog" },
      { ItemId: 5, title: "Our Brand" }
    ]
  },
  {
    headingId: 3,
    heading: "Staples Corporate Solution",
    Items: [
      { ItemId: 1, title: "Office Products" },
      { ItemId: 2, title: "Technology Products" },
      { ItemId: 3, title: "Furniture" },
      { ItemId: 4, title: "Facilites" },
      { ItemId: 5, title: "Promotional Products" }
    ]
  },
  {
    headingId: 4,
    heading: "New Customer",
    Items: [
      { ItemId: 1, title: "Become a Customer" },
      { ItemId: 2, title: "Govt. Customers" },
      { ItemId: 3, title: "Memberships" }
    ]
  },
  {
    headingId: 5,
    heading: "Join the Staples Team",
    Items: [{ ItemId: 1, title: "Careers" }]
  }
];
const footerSocialIcons = [
  { IconId: 1, title: "Linkdin", file: "linkedin.png" },
  { IconId: 2, title: "Twitter", file: "twitter.png" },
  { IconId: 3, title: "Facebook", file: "facebook.png" },
  { IconId: 4, title: "Youtube", file: "youtube.png" },
  { IconId: 5, title: "Instagram", file: "insta.png" }
];
const IMAGE_PATH = "assets/images/icons/";
export { footerData, footerSocialIcons, IMAGE_PATH };
