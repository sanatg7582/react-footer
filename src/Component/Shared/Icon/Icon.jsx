import React from "react";
import PropTypes from "prop-types";
import styles from "./Icon.module.scss";
import { IMAGE_PATH } from "../../../FooterData";
/**
 * Icon function return Icon anywhare where you want
 * @param {string} fileName img of src isrequired
 * @param {string} altTitle img of alt isrequired
 * return img element
 */
const Icon = props => {
  const { fileName, altTitle } = props;
  return (
    <img
      className={styles.iconImage}
      src={`${IMAGE_PATH}${fileName}`}
      alt={altTitle}
    />
  );
};
// propsType validation here
Icon.propTypes = {
  fileName: PropTypes.string.isRequired,
  altTitle: PropTypes.string.isRequired
};
export default Icon;
