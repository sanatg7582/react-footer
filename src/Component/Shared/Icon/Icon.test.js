import React from "react";
import { shallow } from "enzyme";
import Icon from "./Icon";
import { IMAGE_PATH } from "../../../FooterData";

describe("renders Icon without crashing ", () => {
  const props = {
    fileName: "",
    altTitle: ""
  };
  const wrapper = shallow(<Icon {...props} />);
  it("renders without crashing", () => {
    expect(wrapper).toBeTruthy();
  });

  it("Image root path should be same", () => {
    expect(wrapper.find("img").prop("src")).toBe(`${IMAGE_PATH}`);
  });

  it("Snapshot test", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
