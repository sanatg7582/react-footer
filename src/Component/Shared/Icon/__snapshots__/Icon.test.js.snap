// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Icon without crashing  Snapshot test 1`] = `ShallowWrapper {}`;
